cmd/image/qcow2ova/get-image.go

```diff
@@ -0,0 +1,85 @@
+package qcow2ova
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+
+	"k8s.io/klog/v2"
+)
+
+// Downloads or copy the image into the target dir mentioned
+func getImage(dir string, src string) (string, error) {
+	dest := path.Join(dir, path.Base(src))
+	if !isURL(src) {
+		if !fileExists(src) {
+			return "", fmt.Errorf("not a valid URL or file does not exist at %s", src)
+		}
+		klog.Infof("Copying %s into %s", src, dest)
+		if err := cp(src, dest); err != nil {
+			return "", err
+		}
+		klog.Infof("Copy Completed!")
+	} else {
+		out, err := os.Create(dest)
+		if err != nil {
+			return "", err
+		}
+		defer out.Close()
+		klog.Infof("Downloading %s into %s", src, dest)
+		resp, err := http.Get(src)
```

Comment:
Should there be a timeout to cancel the download? Also we should check for HTTP response code to bail out on server side error.

Reply:
make sense, what timeout value can make sense? 🤔

Reply:
good catch! somehow I forgot to handle the nonsuccessful cases, will address them with more unit tests